src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_multi.h

```diff
@@ -0,0 +1,251 @@
+      typename vector_like::value_type d1, d2;
+      d1 = std::abs(filtration[i - 1] - to_project);
+      d2 = std::abs(filtration[i] - to_project);
+      x[parameter] = d1 < d2 ? i - 1 : i;
+    }
+  }
+}
+
+/**
+ * \brief Pushes all of the filtration values of a simplextree onto a grid, c.f. \ref find_coordinates.
+ * \ingroup multiparameter
+ * \param st_multi Multiparameter simplex tree to squeeze on the grid.
+ * \param grid The multiparameter grid. A vector of size `num_parameters`,
+ * whose elements are the elements of the grid for this axis.
+ * \param coordinate_values If set to true the filtration values will be turned into coordinates in this grid
+ * instead of points in this grid.
+ * */
+template <class simplextree_multi>
+void squeeze_filtration(simplextree_multi& st_multi, const multi_filtration_grid &grid, bool coordinate_values = true) {
+  static_assert(simplextree_multi::Options::is_multi_parameter, "Only works for multiparameter simplextrees.");
+  auto num_parameters = static_cast<unsigned int>(st_multi.get_number_of_parameters());
+  if (grid.size() != num_parameters) {
+    throw std::invalid_argument("Grid and simplextree do not agree on number of parameters.");
+  }
+  for (const auto &simplex_handle : st_multi.complex_simplex_range()) {
+    auto &simplex_filtration = st_multi.filtration_mutable(simplex_handle);
+    find_coordinates(simplex_filtration,grid);  // turns the simplexfiltration into coords in the grid
+    if (!coordinate_values) {
+      for (auto parameter = 0u; parameter < num_parameters; parameter++)
+        simplex_filtration[parameter] = grid[parameter][simplex_filtration[parameter]];
+    }
+  }
+  return;
+}
+
+// retrieves the filtration values of a simplextree. Useful to generate a grid.
+/**
+ * \brief Retrieves all of the filtration values, for each simplex dimension, of the simplextree. 
+ * Useful to generate grids. 
+ * \ingroup multiparameter 
+ * \param st_multi Simplextree on which filtration values are exctracted.
+ * \param degrees Only the simpleces of these dimension will be taken into account. Useful for, e.g., Rips
+ * filtrations.
+ * */
+template <class simplextree_multi>
+std::vector<multi_filtration_grid> get_filtration_values(simplextree_multi& st_multi, const std::vector<int> &degrees) {
+  static_assert(simplextree_multi::Options::is_multi_parameter, "Only works for multiparameter simplextrees.");
+  int num_parameters = st_multi.get_number_of_parameters();
+  std::vector<multi_filtration_grid> out(degrees.size(), multi_filtration_grid(num_parameters));
+  std::vector<int> degree_index(degrees.size());
+  int count = 0;
+  for (auto degree : degrees) {
+    degree_index[degree] = count++;
+    out[degree_index[degree]].reserve(st_multi.num_simplices());
+  }
+
+  for (const auto &simplex_handle : st_multi.complex_simplex_range()) {
+    const auto filtration = st_multi.filtration(simplex_handle);
+    const auto degree = st_multi.dimension(simplex_handle);
```

Comment:
There is also `for_each_simplex` which gives you the dimension for free, but your code is fine.